### PR TITLE
Implement opentelemetry

### LINKED ISSUE #32: Consider Telemetry and Instrumentation
**Is your feature request related to a problem? Please describe.**
While not wanting to prematurely optimize, good telemetry is invaluable both in debugging and in operating software generally. Accordingly, it would be good to give some thought now to how we'd like to implement this so we don't have to go back and retrofit a bunch of it after-the-fact.

**Describe the solution you'd like**
An OpenTelemetry-based solution seems best since it is an emerging industry standard and should allow instance operators to integrate with hosted/paid solutions like HoneyComb as well as running a local collector setup for debugging purposes.

**Describe alternatives you've considered**
Console/file-based logging, or a more dedicated solution like Seq or Azure App Insights could be an option, but get expensive quickly. It is possible to run a basic self-hosted OT collector via Docker Compose. I may be wrong but I believe those solutions are beginning to support OpenTelemetry anyway.

### PR DESCRIPTION
**Pull request by edmistond**
# Pull Request Template

## Description

Implements basic OpenTelemetry support.

Additionally, I created a "Solution Items" folder and moved all the random project files (gitignore, nuget config, etc) into it to make them easily accessible in Visual Studio.

Note that currently the OpenTelemetry packages are 1.0.0 *prerelease* versions, so we'll want to track that and upgrade when they're released.

We probably need to give some additional thought to configuration and I will open an issue for that. Once we sort out configuration I would also like to make the OTel support, as well as the exporter, be configurable on/off. Currently it should be both writing to console *and* attempting send to the default collector at localhost:4318/ which is the default; I haven't been able to test this yet since getting a local OpenTelemetry stack with Jaeger and Zipkin running is surprisingly non-trivial due to poor documentation. I am tempted to set up a collector (easy and well documented) and test shipping to the Honeycomb free tier.

Fixes #32 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Configured OTel in the WebApp program.cs, tested that I get console logging for OTel when starting up the application. Also tried hitting the webfinger URL which confirmed that OTel logs activity data when hitting a URL. This is all using preconfigured logging, no custom spans or activities.

**Test Configuration**:
* Hardware: Ryzen 7 2700, Windows 11 22H2
* Toolchain: dotnet 7, JetBrains Rider

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by benrick** [comment]
Awesome!

Regarding configuration, I've been thinking on that as well. Let's open a discussion or issue on that. 

Regarding solution files, I did the same in a branch, I'll back mine out and keep yours! 

I'll try to review this tomorrow, but anyone else in here wanting to can put their review in as well!

File: Smilodon.sln [hunk 1]
```diff
@@ -17,6 +17,16 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ActivityPub.Domain", "src\A
...
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Domain", "src\Domain\Domain.csproj", "{46FC91AA-B4F6-4339-A414-5126474048C6}"
 EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "SolutionItems", "SolutionItems", "{98B687E0-157E-4EF0-8482-6E6EDF46E034}"
+	ProjectSection(SolutionItems) = preProject
+		.editorconfig = .editorconfig
+		.gitattributes = .gitattributes
+		.gitignore = .gitignore
+		Directory.Packages.props = Directory.Packages.props
+		nuget.config = nuget.config
+		README.md = README.md
```
Comment (benrick) [inline Smilodon.sln:27]: Good initial set.

File: src/WebApp/Program.cs [hunk 2]
```diff
@@ -1,12 +1,35 @@
...
+using OpenTelemetry.Exporter;
+using OpenTelemetry.Resources;
+using OpenTelemetry.Trace;
 using Smilodon.Infrastructure.Persistence;
 using Smilodon.WebApp.Api.Admin;
 using Smilodon.WebApp.Api.Webfinger;
 
 var builder = WebApplication.CreateBuilder(args);
 
+// this should really be in the app config instead...
```
Comment (benrick) [inline src/WebApp/Program.cs:11]: Agreed! 🤣 Let's remember to add this when we get configuration going.

File: src/WebApp/Program.cs [hunk 3]
```diff
@@ -1,12 +1,35 @@
...
+        .AddOtlpExporter(opt =>
+        {
+            opt.Protocol = OtlpExportProtocol.HttpProtobuf;
+        })
+        .AddSource(serviceName)
+        .SetResourceBuilder(ResourceBuilder.CreateDefault()
+            .AddService(serviceName: serviceName, serviceVersion: serviceVersion))
+        .AddHttpClientInstrumentation()
+        .AddAspNetCoreInstrumentation()
+        .AddSqlClientInstrumentation();
```
Comment (benrick) [inline src/WebApp/Program.cs:30]: Good for now, but we'll want this in config as well.

**Review by benrick — APPROVED** [review]
Yeah, I can see why this made you want to get configuration sorted. Yeah, we'll need to set that up soon. Looks straight out of the docs, and can should be configurable. 👍

File: Smilodon.sln (hunk 1 above)
Comment (kirkbrauer) [reply Smilodon.sln:27]: Didn't even realize you could do this. Great idea!

**Pull request merged**